File: application/src/test/java/com/smtm/application/security/FakeAuthorization.java
package com.smtm.application.security;

import static com.smtm.security.token.TokenKt.tokenOf;
import java.util.ArrayList;
import java.util.List;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import com.smtm.security.api.Authorization;
import com.smtm.security.token.Token;

public class FakeAuthorization implements Authorization {

    private final List<String> validTokens;

    public FakeAuthorization() {
        this.validTokens = new ArrayList<>();
    }

    @Nullable
    @Override
    public Token authorize(@NotNull String token) {
        if (validTokens.contains(token)) {
            return tokenOf(token, 1);
        }

        return null;
    }

    void addValidTokens(List<String> tokens) {
        this.validTokens.addAll(tokens);
    }
}

File: application/src/test/kotlin/com/smtm/application/security/FakeAuthorization.kt
package com.smtm.application.security

import com.smtm.security.api.Authorization
import com.smtm.security.api.AccessToken

class FakeAuthorization : Authorization {

    val validTokens: MutableList<String> = mutableListOf()

    override fun authorize(token: String): AccessToken? = token
        .takeIf { validTokens.contains(it) }
        ?.let { accessTokenOf(it) }
}
